spec/facets/DatasetLifecycleStateDatasetFacet.json

```diff
@@ -0,0 +1,58 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$id": "https://openlineage.io/spec/facets/1-0-0/DatasetLifecycleDatasetFacet.json",
+  "$defs": {
+    "DatasetLifecycleStateDatasetFacet": {
+      "allOf": [{
+        "$ref": "https://openlineage.io/spec/1-0-2/OpenLineage.json#/$defs/DatasetFacet"
+      }, {
+        "type": "object",
+        "properties": {
+          "lifecycleState": {
+            "description": "The lifecycle state of a dataset.",
+            "type": "string",
+            "enum": ["ALTERED", "CREATED", "DROPPED", "OVERWRITTEN", "RENAMED", "TRUNCATED"]
+          },
+          "previousIdentifier": {
+            "description": "Previous name of the dataset in case of renaming it.",
+            "type": "object",
+            "properties": {
+              "name": {
+                "documentation": "Previous dataset name.",
+                "type": "string"
+              },
+              "namespace": {
+                "documentation": "Previous dataset namespace.",
+                "type": "string"
+              }
+            },
+            "required": ["name", "namespace"]
+          },
+          "currentIdentifier": {
+            "description": "Current name of the dataset in case of renaming it.",
+            "type": "object",
+            "properties": {
+              "name": {
+                "documentation": "Current dataset name.",
+                "type": "string"
+              },
+              "namespace": {
+                "documentation": "Current dataset namespace.",
+                "type": "string"
+              }
+            },
+            "required": ["name", "namespace"]
+          }
```

Comment:
I'd probably reduce this down to only `previousIdentifier`. The current identifier should be captured in the `Dataset` name and namespace already. There's also no restriction here to require a user to record both the `previousIdentifier` and the `currentIdentifier`, so if we rely on both being present, a user may accidentally record only one. Reducing this down to only `previousIdentifier` and using the `Dataset.name` as the current identifier reduces room for error.